drivers/adc/adc_stm32.c

```diff
@@ -805,18 +805,16 @@ static void adc_stm32_setup_channels(const struct device *dev, uint8_t channel_i
 {
 	const struct adc_stm32_cfg *config = dev->config;
 #ifdef CONFIG_SOC_SERIES_STM32G4X
-	ADC_TypeDef *adc = config->base;
-
```

Comment:
Another way to suppress the warning would be to do `(void)adc;` here, or convert the `#if` to conventional `if` (when always false, it will get optimised out by the compiler anyway).

No strong preference on any of these approaches, however.